AK/Utf16View.cpp

```diff
@@ -39,17 +39,43 @@ static ErrorOr<Utf16Data> to_utf16_impl(UtfViewType const& view)
 
 ErrorOr<Utf16Data> utf8_to_utf16(StringView utf8_view)
 {
-    return to_utf16_impl(Utf8View { utf8_view });
+    return utf8_to_utf16(Utf8View { utf8_view });
 }
 
 ErrorOr<Utf16Data> utf8_to_utf16(Utf8View const& utf8_view)
 {
-    return to_utf16_impl(utf8_view);
+    // All callers want to allow lonely surrogates, which simdutf does not permit.
+    if (!utf8_view.validate(Utf8View::AllowSurrogates::No)) [[unlikely]]
+        return to_utf16_slow(utf8_view);
+
+    Utf16Data utf16_data;
+
+    TRY(utf16_data.try_resize(simdutf::utf16_length_from_utf8(
+        reinterpret_cast<char const*>(utf8_view.bytes()),
+        utf8_view.byte_length())));
+
+    [[maybe_unused]] auto result = simdutf::convert_utf8_to_utf16(
```

Comment:
if you wanted to actually use these values, we could put a VERIFY or (keeping the [[maybe_unused]]) ASSERT that the number of bytes transmuted is as expected.

Reply:
(Side note, usually an ASSERT macro would "reference" the variables it captures so that we don't have to [[maybe_unused]] all of the debug-only things)

something like:
```c++
#if NDEBUG
template <typename... Args>
constexpr void yolo(Args &&...) { }

#define ASSERT(...) true ? static_cast<void>(0) : yolo(__VA_ARGS__)
#endif
```